src/UglyToad.PdfPig/Function/PdfFunction.cs

```diff
@@ -0,0 +1,224 @@
+﻿namespace UglyToad.PdfPig.Function
+{
+    using System;
+    using System.Linq;
+    using UglyToad.PdfPig.Tokenization.Scanner;
+    using UglyToad.PdfPig.Tokens;
+
+    /// <summary>
+    /// PdfFunction
+    /// </summary>
+    public abstract class PdfFunction
+    {
+        /// <summary>
+        /// The function dictionary.
+        /// </summary>
+        public DictionaryToken FunctionDictionary { get; }
+
+        /// <summary>
+        /// (Required) An array of 2 × m numbers, where m shall be the number of input values. For each i from 0 to m − 1,
+        /// Domain2i shall be less than or equal to Domain2i+1, and the ith input value, xi, shall lie in the interval
+        /// Domain2i ≤ xi ≤ Domain2i+1. Input values outside the declared domain shall be clipped to the nearest boundary value.
+        /// </summary>
+        public ArrayToken Domain { get; }
```

Comment:
Do we want to convert to either a `float[]/double[]/decimal[]/NumericToken[]` at the point this type is created rather than exposing the raw `ArrayToken` which needs casting and error handling every time?